### PR TITLE
Document the --code-highlight default

### PR DESCRIPTION
**Pull request by hugovk**
Recently had to look this up in the source code.

So include in help text it shows during `pytest --help` and on https://docs.pytest.org/en/7.1.x/reference/reference.html#command-line-flags

---

Whilst looking at the the help texts at https://docs.pytest.org/en/7.1.x/reference/reference.html#command-line-flags, there's quite a bit of variation in case and whether to use a full stop after a single or multiple sentence:

* Sentence case
* Sentence case.
* Sentence case. Sentence case.
* Sentence case. Sentence case
* lower case
* lower case.
* lower case. Sentence case
* lower case. Sentence case.

I can update them if you have a preference? 

I suggest: use sentence case, no full stop for single sentence. If there's more than one sentence, end each with a full stop.

### CODE REVIEW

**Comment by nicoddemus** [comment]
> I suggest: use sentence case, no full stop for single sentence. If there's more than one sentence, end each with a full stop.

Just make sure I got it right, do you mean?

* `one sentence help text`
* `first line of help.\nsecond line of help.`

**Comment by hugovk** [comment]
I was suggesting:

* `One sentence help text`
* `First sentence of help. Second sentence of help.`

And not inserting any `\n`, let the terminal and argument parser wrap as usual.

**Comment by nicoddemus** [comment]
👍 from me 😁

**Comment by hugovk** [comment]
Updated!

File: src/_pytest/helpconfig.py [hunk 1]
```diff
@@ -83,16 +83,17 @@ def pytest_addoption(parser: Parser) -> None:
...
-        "Defaults to 'pytestdebug.log'.",
+        "Default: pytestdebug.log.",
     )
     group._addoption(
         "-o",
         "--override-ini",
         dest="override_ini",
         action="append",
-        help='override ini option with "option=value" style, e.g. `-o xfail_strict=True -o cache_dir=cache`.',
+        help='override ini option with "option=value" style, '
```
Comment (nicoddemus) [inline src/_pytest/helpconfig.py:95]:
```suggestion
        help='Override ini option with "option=value" style, '
```

File: src/_pytest/logging.py [hunk 2]
```diff
@@ -228,7 +228,7 @@ def add_option_ini(option, dest, default=None, type=None, **kwargs):
         default=None,
         metavar="LEVEL",
         help=(
-            "level of messages to catch/display.\n"
+            "Level of messages to catch/display.\n"
```
Comment (nicoddemus) [inline src/_pytest/logging.py:231]:
```suggestion
            "Level of messages to catch/display."
```

File: src/_pytest/mark/__init__.py [hunk 3]
```diff
@@ -96,7 +96,7 @@ def pytest_addoption(parser: Parser) -> None:
         dest="markexpr",
         default="",
         metavar="MARKEXPR",
-        help="only run tests matching given mark expression.\n"
+        help="Only run tests matching given mark expression.\n"
```
Comment (nicoddemus) [inline src/_pytest/mark/__init__.py:99]:
```suggestion
        help="Only run tests matching given mark expression. "
```

Comment (nicoddemus) [reply src/_pytest/mark/__init__.py:99]: I understand we want to remove the `\n` and let the terminal/formatter handle new lines, correct? If not please disregard.

**Review by nicoddemus — CHANGES_REQUESTED** [review]
Looks great, I found a few places that seem to have been missed!

Besides that, I think we should add a CHANGELOG entry, `9883.improvement.rst`:

```rst
Normalize the help description of all command-line options.
```

I figure it is possible that someone is parsing/searching that text using some tool, so it is good to just mention it in the changelog as a heads-up. 👍

File: src/_pytest/logging.py [hunk 4]
```diff
@@ -228,7 +228,7 @@ def add_option_ini(option, dest, default=None, type=None, **kwargs):
...
             "Not set by default, so it depends on the root/parent log handler's"
```
Comment (hugovk) [inline src/_pytest/logging.py:232]:
```suggestion
            " Not set by default, so it depends on the root/parent log handler's"
```

File: src/_pytest/mark/__init__.py (hunk 3 above)
Comment (hugovk) [reply src/_pytest/mark/__init__.py:99]: Yes. I had decided to _not_ add any new ones, and leave the existing ones in, but it's a good idea to remove those too, there's not many.

Comment (nicoddemus) [reply src/_pytest/mark/__init__.py]: I see, thanks!

**Review by nicoddemus — APPROVED** [review]
Thanks @hugovk!

**Pull request merged**